src/sentry/snuba/outcomes.py

```diff
@@ -346,15 +364,23 @@ def __init__(
         for key in self.query_groupby:
             self.group_by.append(Column(key))
 
-        self.conditions = self.get_conditions(query, params)
+        condition_data = {
+            "outcome": outcome,
+            "key_id": key_id,
+            "category": category,
+            "reason": reason,
+        }
+        self.conditions = self.get_conditions(condition_data, params)
 
-    def get_conditions(self, query: QueryDict, params: Mapping[Any, Any]) -> List[Any]:
+    def get_conditions(self, query: Mapping[str, Any], params: Mapping[Any, Any]) -> List[Any]:
         query_conditions = [
             Condition(Column("timestamp"), Op.GTE, self.start),
             Condition(Column("timestamp"), Op.LT, self.end),
         ]
         for filter_name in DIMENSION_MAP:
-            raw_filter = query.getlist(filter_name, [])
+            raw_filter = query.get(filter_name, []) or []
+            if not isinstance(raw_filter, list):
+                raw_filter = [raw_filter]
```

Comment:
`query.getlist()` always returns a list. Adding a 'cast' felt like the best way to preserve the behavior.